### ISSUE TITLE
GPG_AGENT_INFO and SSH_AUTH_SOCK are not set globally with gnome 3.16 keyring

### ISSUE THREAD

**Issue by DamienCassou** [NixOS/nixpkgs#7891]
I recently switched to gnome 3.16 under nixos. To do that, I changed my `configuration.nix`:

```nix
{
  services.xserver.desktopManager.gnome3.sessionPath = with pkgs.gnome3_16; [ gpaste ];
  environment.gnome3.packageSet = pkgs.gnome3_16;
  nixpkgs.config.packageOverrides = pkgs: {
    gtk3 = pkgs.gtk3_16;
    gnome3 = pkgs.gnome3_16;
  };
}
```

Unfortunately, the gnome terminal does have neither the `GPG_AGENT_INFO` nor the `SSH_AUTH_SOCK` environment variables set anymore. As soon as I set them manually with

```bash
export GPG_AGENT_INFO=/run/user/1000/keyring/gpg:0:1
export SSH_AUTH_SOCK=/run/user/1000/keyring/ssh
```

things work fine.

It was working fine with Gnome 3.12.

**Comment by DamienCassou** [comment]
@lethalman might be interesting for you

**Comment by lethalman** [comment]
GNOME packaging drives people crazy... :P I'll see what's happening here,
thanks for testing it!

On Mon, May 18, 2015 at 6:55 PM, Damien Cassou <notifications@github.com>
wrote:

> Assigned #7891 <https://github.com/NixOS/nixpkgs/issues/7891> to
> @lethalman <https://github.com/lethalman>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/NixOS/nixpkgs/issues/7891#event-307590182>.
>



-- 
NixOS Linux <http://nixos.org>

**Comment by lethalman** [comment]
See http://lists.gnupg.org/pipermail/gnupg-devel/2014-August/028689.html . So it's upstream that has dropped support for gpg somehow. Closing.

**Comment by DamienCassou** [comment]
$ pass show -c inria.fr
gpg: WARNING: The GNOME keyring manager hijacked the GnuPG agent.
gpg: WARNING: GnuPG will not work properly - please configure that tool to not interfere with the GnuPG system!
Copied inria.fr to clipboard. Will clear in 45 seconds.
cassou@luz3:~$ gnome-shell --version
GNOME Shell 3.16.1

**Comment by lethalman** [comment]
It's weird. I was able to `gpg2 -d` something and pinentry showed... gpg itself started `gpg-agent --server`.  So gnome-keyring correctly is not involved but rather pinentry. What else do you have in your configuration? Like disabled ssh agent or such?

Do you have gpg-agent in your PATH?

**Comment by DamienCassou** [comment]
```
$ which gpg-agent
/home/cassou/.nix-profile/bin/gpg-agent
cassou@luz3:~$ realpath /home/cassou/.nix-profile/bin/gpg-agent
/nix/store/ll4lz7ycs2cw9kki09wpxfr670jw0pa8-gnupg-2.0.27/bin/gpg-agent
$ gpg-agent --version
gpg-agent (GnuPG) 2.0.27
libgcrypt 1.6.3
Copyright (C) 2015 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
$ 
```

I have `gnupg` in my system.nix, `openssh.enable=true` in my `configuration.nix`

**Comment by DamienCassou** [comment]
Do you also face the hijacking warnings?

```
$ gpg2 --decrypt /tmp/test.txt.gpg 

You need a passphrase to unlock the secret key for
user: "Damien Cassou <damien@cassou.me>"
4096-bit RSA key, ID 0DB0F7D4, created 2015-04-17 (main key ID F5DCE914)

gpg: WARNING: The GNOME keyring manager hijacked the GnuPG agent.
gpg: WARNING: GnuPG will not work properly - please configure that tool to not interfere with the GnuPG system!
gpg: encrypted with 4096-bit RSA key, ID 0DB0F7D4, created 2015-04-17
      "Damien Cassou <damien@cassou.me>"
```

**Comment by lethalman** [comment]
No I don't. Maybe there's something with seahorse too, I don't know it's a complex world. We better look for similar problems on the internet.

**Comment by bendlas** [comment]
Those hijacking warnings are the kind of inter-community headbutting, that makes everybody flock towards free software⸮
https://bbs.archlinux.org/viewtopic.php?id=182889
https://bugs.gnupg.org/gnupg/issue1656

It can be disabled pretty easily: https://gebner.org/how-to-get-rid-of-the-the-gnome-keyring-manager-hijacked-the-gnupg-agent-warning.html

Maybe it would be worthwhile to let nixos users chose between gnome's agent (while disabling the warning) and gpg's agent.

**Comment by lethalman** [comment]
@bendlas if I knew how gnome-keyring worked :D As you see from the comments above, I don't have this problem, thought I have a similar system configuration to @DamienCassou

**Comment by DamienCassou** [comment]
@lethalman do you have a file `~/.config/autostart/gnome-keyring-ssh.desktop`? See http://askubuntu.com/questions/162850/how-to-disable-the-keyring-for-ssh-and-gpg

**Comment by ellis** [comment]
The information I'm adding here probably isn't relevant, but I came across this issue trying to troubleshoot a problem I'm having with gpg-agent.  I've been running my own expressions for lxqt for a long time, but since I updated the unstable channel a few weeks ago, GPG_AGENT_INFO and SSH_AUTH_SOCK are no longer automatically set for me either.  If I remember correctly, gpg-agent used to be started from the X startup scripts when the gpg-agent was enabled in configure.nix.
